packages/angular_devkit/build_angular/src/builders/dev-server/specs/live-reload_spec.ts

```diff
@@ -22,10 +24,20 @@ interface ProxyInstance {
   url: string;
 }
 
-let proxyPort = 9100;
-function createProxy(target: string, secure: boolean, ws = true): ProxyInstance {
-  proxyPort++;
+function findFreePort(): Promise<number> {
+  return new Promise<number>((resolve, reject) => {
+    const server = createServer();
+    server.once('listening', () => {
+      const port = (server.address() as AddressInfo).port;
+      server.close((e) => (e ? reject(e) : resolve(port)));
+    });
+    server.once('error', (e) => server.close(() => reject(e)));
+    server.listen();
+  });
+}
 
+async function createProxy(target: string, secure: boolean, ws = true): Promise<ProxyInstance> {
+  const proxyPort = await findFreePort();
```

Comment:
Can this be zero and then use `(server.address() as AddressInfo).port` on the proxy server below to get the port to use in the test?

Reply:
That's unfortunate.